src/ansys/fluent/core/session.py

```diff
@@ -258,60 +259,21 @@ def __exit__(self, exc_type: Any, exc_val: Any, exc_tb: Any):
         logger.debug("session.__exit__() called")
         self.exit()
 
-    def upload(self, file_path: str, remote_file_name: Optional[str] = None):
-        """Uploads a file on the server."""
-        if not self._uploader:
-            self._uploader = _Uploader(self.fluent_connection._remote_instance)
-        return self._uploader.upload(file_path, remote_file_name)
-
-    def download(self, file_name: str, local_file_path: Optional[str] = None):
-        """Downloads a file from the server."""
-        if not self._uploader:
-            self._uploader = _Uploader(self.fluent_connection._remote_instance)
-        return self._uploader.download(file_name, local_file_path)
-
-
-class _Uploader:
-    """Instantiates a file uploader and downloader to have a seamless file reading /
-    writing in the cloud particularly in Ansys lab . Here we are exposing upload and
-    download methods on session objects. These would be no- ops if PyPIM is not
-    configured or not authorized with the appropriate service. This will be used for
-    internal purpose only.
-
-    Attributes
-    ----------
-    pim_instance: PIM instance
-        Instance of PIM which supports upload server services.
-
-    file_service: Client instance
-        Instance of Client which supports upload and download methods.
-
-    Methods
-    -------
-    upload(
-        file_path, remote_file_name
-        )
-        Upload a file to the server.
-
-    download(
-        file_name, local_file_path
-        )
-        Download a file from the server.
-    """
-
-    def __init__(self, pim_instance):
-        self.pim_instance = pim_instance
-        self.file_service = None
+    def _file_service(self):
+        """File upload and download service."""
+        pim_instance = self.fluent_connection._remote_instance
+        file_service = None
```

Comment:
@hpohekar Should we be caching this file service or executing all this code each time?

Reply:
No. Right, `_file_service` is called twice. (in upload and download separately). We can combine it in one method and we can handle upload and download by passing an argument then we will call this single method in `session.upload()` and `session.download()` methods. I got your point, I will update this soon. Thank you.

Reply:
Implemented `_upload_download_helper`. We are going to do similar implementation for `_pypim_upload_download_helper`.

Reply:
moved `_file_service` to `__init__` and we are creating it only once and using in all required methods.